src/openassetio-python/tests/cmodule/resources/_openassetio_test/errorsTest.cpp

```diff
@@ -401,6 +408,20 @@ void registerExceptionThrower(py::module_& mod) {
       py::arg("exceptionThrower"), py::arg("catchExceptionName"),
       py::call_guard<py::gil_scoped_release>{});
 
+  mod.def(
+      "throwPythonExceptionAndCatchAsStdException",
+      [](ExceptionThrower& exceptionThrower) {
+        try {
+          exceptionThrower.throwFromOverride();
+        } catch (const std::exception&) {  // NOLINT(*-empty-catch)
+        }
+        try {
+          exceptionThrower.throwFromOverride();
+        } catch (const std::runtime_error&) {  // NOLINT(*-empty-catch)
+        }
```

Comment:
[Question] Again just to make sure I understand. The idea is that an uncaught exception here fails the test implicitly because it falls through the catch and escapes as an uncaught exception, causing a general error?

Might be an idea to just note that in the catch statements `//uncaught exception causes error` or something to jog the mind.

Reply:
Added a comment in 743e0b0b